File: foodapp/src/main/java/com/kspk/controller/CartController.java
package com.kspk.controller;

import com.kspk.DTOs.CartRequest;
import com.kspk.entity.Cart;
import com.kspk.service.CartService;
import lombok.AllArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/cart")
@AllArgsConstructor
public class CartController {

    @Autowired
    private final CartService cartService;


    @PostMapping
    public ResponseEntity<?> addToCart(@RequestBody CartRequest request){
        String foodId=request.getFoodId();
        if(foodId==null || foodId.isEmpty()){
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Food id is required");
        }

        return ResponseEntity.ok().body(cartService.addToCart(request));
    }

    @GetMapping
    public ResponseEntity<?> getCart(){
        return ResponseEntity.ok().body(cartService.getCart());
    }

    @DeleteMapping
    public ResponseEntity<?> clearCart(){
        cartService.clearCart();
       return ResponseEntity.ok().build();
    }

    @PostMapping("/remove")
    public ResponseEntity<?> removeCart(@RequestBody CartRequest request){
        String foodId=request.getFoodId();
        if(foodId==null || foodId.isEmpty()){
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Food id is required");
        }

        return ResponseEntity.ok().body(cartService.removeFromCart(request));
    }
}

File: food-app-backend-js/src/controllers/cart.controller.js
import cartService from '../services/cart.service.js';

class CartController {
  async addToCart(req, res) {
    const { foodId } = req.body;
    if (!foodId) return res.status(400).send('Food id is required');

    try {
      const response = await cartService.addToCart(req.userId, req.body);
      res.status(200).json(response);
    } catch (error) {
      res.status(500).send('Failed to add item to cart.');
    }
  }

  async getCart(req, res) {
    try {
      const response = await cartService.getCart(req.userId);
      res.status(200).json(response);
    } catch (error) {
      res.status(500).send('Failed to retrieve cart.');
    }
  }

  async clearCart(req, res) {
    try {
      await cartService.clearCart(req.userId);
      res.status(200).end();
    } catch (error) {
      res.status(500).send('Failed to clear cart.');
    }
  }

  async removeCart(req, res) {
    const { foodId } = req.body;
    if (!foodId) return res.status(400).send('Food id is required');

    try {
      const response = await cartService.removeFromCart(req.userId, req.body);
      if (response === null) {
        return res
          .status(200)
          .json({ message: 'Item not found or quantity was already zero.' });
      }
      res.status(200).json(response);
    } catch (error) {
      res.status(404).send({ message: error.message });
    }
  }
}

export default new CartController();
